src/app/chat.service.ts

```diff
@@ -24,6 +24,11 @@ export class ChatService {
           if (!repeatedMessage) {
             this.messages.push(response.data);
             this.messages.sort((a, b) => a.id - b.id);
+            this.socket.emit('message-ack', {
+              clientId: this.userService.user.id,
+              groupId: this.currentGroup.id,
+              messageId: this.messages[this.messages.length - 1].id
```

Comment:
ส่งด้วย id ที่ได้จาก response.data จะปลอดภัยกว่านะ เผื่อมีฟังก์ชันอื่นทำงานแล้วอัพเดท this.messages ก้อนนี้พอดีแล้วลำดับมันจะเปลี่ยนไปแล้วอ่ะ